project/default.nix

```diff
@@ -133,30 +140,78 @@ let
-      ({ package = p: p.${name}; } // config)
-  ) android;
-
-  ios = mapAttrs (name: config:
-    let ghcIosArm64 = this.ghcIosArm64.override { overrides = overrides'; };
-    in (this.iosWithHaskellPackages ghcIosArm64).buildApp
-      ({ package = p: p.${name}; } // config)
-  ) ios;
-
-  reflex = this;
-})
+  prj = mapAttrs mkPkgSet shells // {
+    shells = mapAttrs (name: pnames:
+      this.workOnMulti' {
+        env = prj.${name}.override { overrides = self: super: nixpkgs.lib.optionalAttrs withHoogle {
+          ghcWithPackages = self.ghcWithHoogle;
+        }; };
+        packageNames = pnames;
+        inherit tools;
+      }
+    ) shells;
+
+    android =
+      mapAttrs (name: config:
+        let
+          ghcAndroidArm64 = this.ghcAndroidArm64.override { overrides = overrides'; };
+          ghcAndroidArmv7a = this.ghcAndroidArmv7a.override { overrides = overrides'; };
+        in (this.androidWithHaskellPackages { inherit ghcAndroidArm64 ghcAndroidArmv7a; }).buildApp
+          ({ package = p: p.${name}; } // config)
+      ) android;
+
+    ios =
+      mapAttrs (name: config:
+        let ghcIosArm64 = this.ghcIosArm64.override { overrides = overrides'; };
+        in (this.iosWithHaskellPackages ghcIosArm64).buildApp
+          ({ package = p: p.${name}; } // config)
+      ) ios;
+
+    reflex = this;
+
+    all = all true;
+  };
+
+  ghcLinks = mapAttrsToList (name: pnames: optionalString (pnames != []) ''
+    mkdir -p $out/${name}
+    ${concatMapStringsSep "\n" (n: ''
+      ln -s ${prj.${name}.${n}} $out/${name}/${n}
+    '') pnames}
+  '') shells;
+  mobileLinks = mobileName: mobile: ''
+    mkdir -p $out/${mobileName}
+    ${concatStringsSep "\n" (mapAttrsToList (name: app: ''
+      ln -s ${app} $out/${mobileName}/${name}
+    '') mobile)}
+  '';
+
+  all = includeRemoteBuilds:
+    let tracedMobileLinks = mobileName: system: mobile:
+      let
+        build = mobileLinks mobileName mobile;
+        msg = ''
+
+
+          Skipping ${mobileName} apps; system is ${this.system}, but ${system} is needed.
+          Use `nix-build -A all` to build with remote machines.
+          See: https://nixos.org/nixos/manual/options.html#opt-nix.buildMachines
+
+        '';
+      in if mobile == {} then ""
+        else if includeRemoteBuilds then build
+          else if system != this.system then builtins.trace msg ""
+            # TODO: This is a bit of a hack. `this.iosSupport` prints
+            # a warning and returns false when *the local system*
+            # doesn't have the SDK. Just because `includeRemoteBuilds`
+            # is off doesn't mean we know this is the system iOS apps
+            # will build on. Nonetheless, it's important not to
+            # evaluate `this.iosSupport` if we don't need to, as it
+            # may `trace` an unnecessary warning.
+            else if system == "x86_64-darwin" -> this.iosSupport then build
```

Comment:
@ryantrinkle This is a little awkward, but I'm not sure what else to do about it besides figuring out how to get the SDK in the nix store properly

Reply:
I think we're gonna need to change how we retrieve the iOS SDK eventually.  However, this is fine for now.